python/lsst/cp/pipe/cpCombine.py

```diff
@@ -341,6 +341,18 @@ def run(self, inputExpHandles, inputScales=None, inputDims=None):
         self.combineHeaders(inputExpHandles, combinedExp,
```

Comment:
I'm a bit confused by this call because we don't do anything with the header this calculates.

I also just looked in combineHeaders and I wonder if we can make use of the `first` and `last` parameters to `merge_headers` -- those are designed to allow you to pick the first DATE-OBS you encounter and the last DATE-END you encounter such that the ObservationGroup code at the end might not be needed. Also we now use DATE-BEG in addition to DATE-OBS.

Reply:
Okay. It's all very confusing and seems to be working differently to what it's documented to do. There is the added confusion that it generates a merged header but then copies everything that got dropped back into the original header but that's exactly what the `first` mode option does in `merge_headers` anyhow, and if the `first` and `last` parameters are used it will automatically makes sure it picks up the first DATE-OBS and the last DATE-END.  The way it's now written you don't want `merged` to exist at all because you are wanting to use the other header you got from the exposure and then copy everything back out of `merged` into the alias.

I think ObservationGroup is used because that can sort the headers into date order before it reads back the start time and end time (which it gets from header translation).